rxandroid/src/test/java/io/reactivex/android/plugins/RxAndroidPluginsTest.java

```diff
@@ -67,32 +72,99 @@ public void resetClearsMainThreadHandler() {
 
     @Test
     public void initMainThreadHandlerCalled() {
-        final AtomicReference<Scheduler> schedulerRef = new AtomicReference<>();
+        final AtomicReference<Callable<Scheduler>> schedulerRef = new AtomicReference<>();
         final Scheduler newScheduler = new EmptyScheduler();
-        RxAndroidPlugins.setInitMainThreadSchedulerHandler(new Function<Scheduler, Scheduler>() {
-            @Override public Scheduler apply(Scheduler scheduler) {
-                schedulerRef.set(scheduler);
-                return newScheduler;
-            }
-        });
+        RxAndroidPlugins
+                .setInitMainThreadSchedulerHandler(new Function<Callable<Scheduler>, Scheduler>() {
+                    @Override public Scheduler apply(Callable<Scheduler> scheduler) {
+                        schedulerRef.set(scheduler);
+                        return newScheduler;
+                    }
+                });
 
-        Scheduler scheduler = new EmptyScheduler();
+        Callable<Scheduler> scheduler = new Callable<Scheduler>() {
+            @Override public Scheduler call() throws Exception {
+                return new EmptyScheduler();
+            }
+        };
         Scheduler actual = RxAndroidPlugins.initMainThreadScheduler(scheduler);
         assertSame(newScheduler, actual);
         assertSame(scheduler, schedulerRef.get());
     }
 
     @Test
-    public void resetClearsInitMainThreadHandler() {
-        RxAndroidPlugins.setInitMainThreadSchedulerHandler(new Function<Scheduler, Scheduler>() {
-            @Override public Scheduler apply(Scheduler scheduler) {
-                throw new RuntimeException();
+    public void resetClearsInitMainThreadHandler() throws Exception {
+        RxAndroidPlugins
+                .setInitMainThreadSchedulerHandler(new Function<Callable<Scheduler>, Scheduler>() {
+                    @Override public Scheduler apply(Callable<Scheduler> scheduler) {
+                        throw new RuntimeException();
+                    }
+                });
+
+        final Scheduler scheduler = new EmptyScheduler();
+        Callable<Scheduler> schedulerCallable = new Callable<Scheduler>() {
+            @Override public Scheduler call() throws Exception {
+                return scheduler;
             }
-        });
+        };
+
         RxAndroidPlugins.reset();
 
-        Scheduler scheduler = new EmptyScheduler();
-        Scheduler actual = RxAndroidPlugins.initMainThreadScheduler(scheduler);
-        assertSame(scheduler, actual);
+        Scheduler actual = RxAndroidPlugins.initMainThreadScheduler(schedulerCallable);
+        assertSame(schedulerCallable.call(), actual);
     }
+
+    @Test
+    public void defaultMainThreadSchedulerIsInitializedLazily() {
+        final Function<Callable<Scheduler>, Scheduler> safeOverride = new Function<Callable<Scheduler>, Scheduler>() {
+            @Override public Scheduler apply(Callable<Scheduler> __) {
```

Comment:
Just to clarify; are you asking that I replace the unnamed parameter: `__` with the named parameter: `scheduler`?